src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineGeneralTest.java

```diff
@@ -640,6 +646,73 @@ public void batchInsertAutocommitTest() throws Exception {
         assertEquals("COL5  ok?", "OLA", query.get(1).get("COL5").toString());
     }
 
+    /**
+     * Tests that on a rollback situation, the prepared statement batches are cleared.
+     *
+     * The steps performed on this test are:
+     * <ol>
+     *     <li>Add batch to transaction and purposely fail to flush</li>
+     *     <li>Ensure the existence of the Exception and rollback transaction</li>
+     *     <li>Flush again successfully an ensure that the DB table doesn't have any rows</li>
+     * </ol>
+     *
+     * This is a regression test.
+     *
+     * @throws DatabaseEngineException If there is a problem on {@link DatabaseEngine} operations.
+     * @since 2.1.12
+     */
+    @Test
+    public void batchInsertRollback() throws DatabaseEngineException {
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        final DbEntity entity = dbEntity()
+                .name("TEST")
+                .addColumn("COL1", INT)
+                .build();
+
+        new MockUp<AbstractDatabaseEngine>() {
+            @Mock
+            public synchronized void flush(final Invocation invocation) throws DatabaseEngineException {
+                if (latch.getCount() == 1) {
+                    throw new DatabaseEngineException("");
+                }
+                invocation.proceed();
+            }
+        };
+
+        DatabaseEngineException expectedException = null;
+
+        engine.addEntity(entity);
+        engine.beginTransaction();
+
+        try {
+            final EntityEntry entry = entry().set("COL1", 1).build();
+
+            engine.addBatch("TEST", entry);
+            engine.flush();
```

Comment:
maybe put a _fail("message")_ after the flush?
the flush shouldn't complete, therefore if it does fail the test